### PR TITLE
Style field validation message

### PR DESCRIPTION
**Pull request by phalkunz**
Issue: https://github.com/silverstripe/silverstripe-admin/issues/160

### CODE REVIEW

File: client/src/components/Form/Form.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ class Form extends Component {
     const className = ['form'];
     if (valid === false) {
       className.push('form--invalid');
+      className.push('was-validated');
```
Comment (flamerohr) [inline client/src/components/Form/Form.js]: what's the purpose of this class? how come it's not BEM?

File: client/src/components/TextField/TextField.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ class TextField extends SilverStripeComponent {
       value: this.props.value,
       placeholder: this.props.placeholder,
       autoFocus: this.props.autoFocus,
+      required: Boolean(this.props.meta && this.props.meta.error),
```
Comment (flamerohr) [inline client/src/components/TextField/TextField.js]: nice, do we need to repeat this for other fields as well?

**Comment by flamerohr** [comment]
I was also curious if we could use something from bootstrap that doesn't require the `:valid`/`:invalid` selector, as they won't work for customised fields like `react-select`

I've found this in the docs for bootstrap4:
> As a fallback, `.is-invalid` and `.is-valid` classes may be used instead of the pseudo-classes for server side validation. They do not require a .was-validated parent class.

Have you tried looking into using that instead @phalkunz ?

**Comment by phalkunz** [comment]
@flamerohr Did you mean using `.is-invalid` class instead of setting `required` prop?

File: client/src/components/TextField/TextField.js (hunk 2 above)
Comment (phalkunz) [reply client/src/components/TextField/TextField.js]: Yes for the fields that are not subclass of `TextField`. Depends on the other comment, I might need to replicate this in other field components.

**Comment by flamerohr** [comment]
that's right :) I wonder if that would mean we don't need to the `noValidate` on the form element anymore

**Comment by phalkunz** [comment]
@flamerohr Good idea. I'll look into fixing on Monday :)

File: client/src/components/Form/Form.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ class Form extends Component {
     );
 
     return (
+      // Needs `noValidate` here in order to prevent browser's message
```
Comment (flamerohr) [inline client/src/components/Form/Form.js]: sweet, don't need `noValidate` anymore :)

File: client/src/components/FieldHolder/FieldHolder.js [hunk 4]
```diff
@@ -115,7 +118,12 @@ function fieldHolder(Field) {
     }
 
     renderField() {
-      const field = <Field {...this.props} />;
+      const props = { ...this.props };
+      const hasErrors = Boolean(this.props.meta && this.props.meta.error);
+      if (hasErrors) {
+        props.extraClass = `${props.extraClass} is-invalid`;
```
[the remark this answers is not in the document]
Comment (flamerohr) [reply client/src/components/FieldHolder/FieldHolder.js]: I'd also recommend replicating the `renderMessage()` logic to the `hasErrors` check:
- checking there is a message
- adding the `meta.touched` and `meta.dirty` checks

so you don't get a red highlighted field with no message

**Pull request merged**